### PR TITLE
143457427 ne update rails api

### PR DESCRIPTION
**Pull request by ski-climb**
@tmikeschu @DavidKnott @ethanbennett Ready for review; 2 reviewers please

This is just the Lot model and its connection to Product.  I'm not completely thrilled that I had to use a callback to generate the UID for the Lot, but I'm absolutely open to suggestions.

### CODE REVIEW

**Review by DavidKnott — APPROVED** [review]
Nice job with the custom validations!

File: spec/models/product_spec.rb [hunk 1]
```diff
@@ -5,4 +5,8 @@
     it { is_expected.to validate_presence_of(:name) }
     it { is_expected.to validate_presence_of(:description) }
   end
+
+  context 'relationships' do
+    it { is_expected.to have_many(:lots) }
```
Comment (tmikeschu) [inline spec/models/product_spec.rb:10]: I know you love your is_expected to, I just have to express my love for SHOULDa matchers 😉

File: app/models/lot.rb [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+  validates_length_of :uid, minimum: 42, maximum: 42
+  validates_uniqueness_of :uid
+  validate :uid_start
+  validate :uid_hex
+  belongs_to :product
+  before_validation :generate_uid
+
+  private 
+
+    def generate_uid
```
Comment (tmikeschu) [inline app/models/lot.rb:11]: Clever!

**Review by tmikeschu — APPROVED** [review]
Nice work, @ski-climb !

**Pull request merged**